SIG documentation (#168)
cilium/community: SIG.md

Reviewer: Side comment, I'm wondering about the role of special-purpose Slack channels for specific topics like #testing, #sig-vendor, #sig-lb and so on which are likely not active enough to require a full process like this, but it can still be useful to have a dedicated channel with a target topic scope. I guess we can find new names / new prefix for them.. just thinking that it may be confusing for contributors to find a channel like #sig-vendor but there is not a formal SIG for that topic, it's just a channel that interested folks find.

Maybe in practice on the way to creating a full SIG we may create channels for some preliminary version of the SIG which can act as the magnet for interested contributors to discuss the topic and figure out if there's enough work to motivate a more formal SIG.

Reviewer 2: Maybe we could add a prefix like `area-` to the ones that aren't SIGs yet to differentiate them. So `sig-scalability` vs `area-datapath`

Reviewer: I like that idea @xmulligan . Happy to make that effective when we merge this PR.

Reviewer 3: I really like this, as it gives both a clear place to go for people who are interested in something, and a clear path for people to formalize something if they are really depending on that functionality.

(For example, we've got a few heavy Service Mesh users that may be interested in being involved with a more formal SIG-ServiceMesh, if we go down that road)_

# Cilium SIGS

The Cilium project has Special Interest Groups, or SIGs, with a common purpose of advancing the project with respect to a specific topic, such as network policy or documentation. Our goal is to enable a distributed decision structure and code ownership, as well as providing focused forums for getting work done, making decisions, and onboarding new contributors. 

Areas covered by SIGs may be vertically focused on particular components or functions, cross-cutting/horizontal, spanning many/all functional areas of the project, or in support of the project itself. Examples:

- Policy
- Scalability
- Docs

> Each SIG must have a charter that specifies its scope (topics, subsystems, code repos, and/or directories), responsibilities, areas of authority, and basic governance. See the SIG charter process for details on how charters are managed. SIGs are relatively free to customize or change how they operate, within some broad guidelines and constraints imposed by cross-project processes (e.g., the release process) and assets (e.g., the cilium repo).

Reviewer 4: Do we want SIGs to have their own governance at this stage? Also unsure about "areas of authority" at this point (whether this isn't already covered by the authority defined by project roles)

Reviewer 2: For basic governance, I meant who the leaders are, but I could also see how that could be confusing.

Reviewer 4: We might also want to flesh out the relationship between project roles (reviewer, committer) and SIGs. For example, I think one of the goals was that people who are aspiring to be reviewers can ask to build up expertise by getting assigned PRs to review within the SIG.

A primary reason that SIGs exist is as forums for collaboration. Much work in a SIG should stay local within that SIG. However, SIGs must communicate in the open, ensure other SIGs and community members can find notes of meetings, discussions, designs, and decisions, and periodically communicate a high-level summary of the SIG's work to the community.

# SIG Roles
## SIG Lead

- Membership tracked in sigs.yaml
- MUST be an odd number of SIG Leads in a group.
- MUST be a reviewer of one or more areas of the project, and should be a subject matter expert in their SIG area.

> - SHOULD define how priorities and commitments are managed. MAY delegate to other leads as needed.

Reviewer 4: I'm not sure that Cilium as an OSS project can define priorities or commitments. It is up to volunteers what they want to spend their time on

- SHOULD drive charter changes (including creation) to get community buy-in but MAY delegate content creation to SIG contributors.

> - MUST identify, track, and maintain the metadata of the SIGs enhancements for the current release and serve as point of contact for the release team, but MAY delegate to other contributors to fulfill these responsibilities.

Reviewer 4: Not sure we have this enhancement metadata (or even a release team in the way that K8s does)

> - MUST organize a main group meeting and make sure sigs.yaml is up to date.

Reviewer 4: I thought we were leaving SIG meetings as optional? i.e. any given SIG can decide whether to have meetings, or just a Slack channel

> - SHOULD facilitate meetings but MAY delegate to other members or future chairs/chairs in training.

Reviewer 4: chair or lead?

- MUST ensure meetings are recorded and at least notes made available.
- MUST coordinate communication and be a connector with other community groups like SIGs and the committers but MAY delegate the actual communication and creation of content to other contributors where appropriate.

### Requirements
Chairs MUST be at least a "member" on our [contributor ladder](https://github.com/cilium/community/blob/main/CONTRIBUTOR-LADDER.md#organization-member) to be eligible to hold a leadership role within a SIG.
SIGs MAY prefer various levels of domain knowledge depending on the role. This should be documented.

### On-boarding and Off-boarding Leads
Chairs MAY decide to step down at anytime and propose a replacement. Use lazy consensus amongst the other Chairs with fallback on majority vote to accept the proposal. The candidate SHOULD be supported by a majority of SIG contributors.
Chairs MAY select additional leads through a super-majority vote amongst Chairs. This SHOULD be supported by a majority of SIG contributors.

## Other Roles

> This governance document outlines the required roles for SIGs; however, SIGs are allowed to operate how they see fit outside of minimum governance requirements, including defining more roles to sustain the group. If a SIG needs to change the Chair position to include or remove duties, this needs to be approved by the committers. Newly created roles that don't assume any responsibility of Chairs should follow the governing processes in the SIG's charter.

Reviewer 4: Not sure we need this at this stage? I would keep it as simple as we can

Author: we can remove it, might not be a bad thing to show that others can pitch in to take on responsibility?

Reviewer 2: Yes, I would leave it for now

# SIG Creation

## Goals
The primary goal of the charters is to define the scope of the SIG within Cilium and how the SIG leaders exercise ownership of these areas by taking care of their responsibilities. A majority of the effort should be spent on these concerns.

> ### SIG Charter Guide

Reviewer 4: We talked about having SIG Leads taking a particular role in reviewing CFPs. I think we should spell that out either here in this document, or in the Charter Guide. 

The intent is offload work from the committers, so they aren't a bottleneck on CFP reviews
 - expectation that SIG leads would be responsible for reviewing CFPs in their area
 - they have a responsibility to bring concerns to committers if they are unsure

I think also we might need to spell out that SIGs don't have additional authority - they are expected to drive towards solutions that can be accepted by the project's committer community (ideally by consensus, but if necessary by a vote)

Author: good point! added in the "SIGs don't have additional authority" near the top of the file

1. Work with the rest of the SIG in question to craft the changes. Make sure to keep the SIG in the loop as discussions progress with the committers.
1. Work with the committers to gain approval. This can simply be submitting a PR and sending a slack message. If more substantial changes are desired it is advisable to socialize those before drafting a PR.
1. The committers will be looking to ensure the scope of the SIG as represented in the charter is reasonable (and within the scope of Cilium) and that processes are fair.

# SIG Governance

To standardize Special Interest Group efforts, create maximum transparency, and route contributors to the appropriate SIG; SIGs should follow these guidelines:

- Have an approved Charter.
- Keep up-to-date meeting notes, linked from the SIG's page in the community repo.
- Ensure related work happens in a project-owned GitHub org and repository.
- Use the forums provided as the primary means of working, communicating, and collaborating, as opposed to private emails and meetings.
- Track and identify all CFPs and other project enhancements related to the SIG area.

## SIG Retirement

> In the event that the SIG is unable to regularly establish consistent quorum or otherwise fulfill its Organizational Management responsibilities

Reviewer 4: Do we have a definition of "quorum" or "Organizational Managament" for a SIG? I would keep this as simple as we can, something like "In the event that a SIG is inactive for 3 or more months, it should be retired"

# Cilium SIGS

The Cilium project has Special Interest Groups, or SIGs, with a common purpose of advancing the project with respect to a specific topic, such as network policy or documentation. Our goal is to enable a distributed decision structure and code ownership, as well as providing focused forums for getting work done, making decisions, and onboarding new contributors. 

Areas covered by SIGs may be vertically focused on particular components or functions, cross-cutting/horizontal, spanning many/all functional areas of the project, or in support of the project itself. Examples:

> - Policy, Scaling, Tetragon, Hubble, Service Mesh

Reviewer 2: ```suggestion

- Policy
- Scalability
- Docs
```

> 

Reviewer 2: ```suggestion
```

> Each SIG must have a charter that specifies its scope (topics, subsystems, code repos, and/or directories), responsibilities, areas of authority, how members and roles of authority/leadership are selected/granted, how decisions are made, and how conflicts are resolved. See the SIG charter process for details on how charters are managed. SIGs are relatively free to customize or change how they operate, within some broad guidelines and constraints imposed by cross-project processes (e.g., the release process) and assets (e.g., the cilium repo).

Reviewer 2: ```suggestion
Each SIG must have a charter that specifies its scope (topics, subsystems, code repos, and/or directories), responsibilities, and areas of authority. See the SIG charter process for details on how charters are managed. SIGs are relatively free to customize or change how they operate, within some broad guidelines and constraints imposed by cross-project processes (e.g., the release process) and assets (e.g., the cilium repo).
```

A primary reason that SIGs exist is as forums for collaboration. Much work in a SIG should stay local within that SIG. However, SIGs must communicate in the open, ensure other SIGs and community members can find notes of meetings, discussions, designs, and decisions, and periodically communicate a high-level summary of the SIG's work to the community.

# SIG Roles
## SIG Lead

> - Membership tracked in sigs.yaml

Reviewer 2: This file is missing from the PR and should be linked here

> - Must be an odd number of SIG Leads in a group

Reviewer 2: ```suggestion
- MUST be an odd number of SIG Leads in a group
```

> - MUST be a review of one or more areas of the project, and should be a subject matter expert in their SIG area

Reviewer 2: ```suggestion
- MUST be a reviewer of one or more areas of the project, and should be a subject matter expert in their SIG area
```

- SHOULD define how priorities and commitments are managed. MAY delegate to other leads as needed.
- SHOULD drive charter changes (including creation) to get community buy-in but MAY delegate content creation to SIG contributors.
- MUST identify, track, and maintain the metadata of the SIGs enhancements for the current release and serve as point of contact for the release team, but MAY delegate to other contributors to fulfill these responsibilities.

> - MUST organize a main group meeting and make sure sigs.yaml is up to date

Reviewer 2: ```suggestion
- MUST organize a main group meeting and make sure sigs.yaml is up to date.
```

- SHOULD facilitate meetings but MAY delegate to other members or future chairs/chairs in training.

> - MUST ensure meetings are recorded and made at least notes made available.

Reviewer 2: ```suggestion
- MUST ensure meetings are recorded and at least notes made available.
```

- MUST coordinate communication and be a connector with other community groups like SIGs and the committers but MAY delegate the actual communication and creation of content to other contributors where appropriate.

### Requirements

> Chairs MUST be at least a "member" on our contributor ladder to be eligible to hold a leadership role within a SIG.

Reviewer 2: ```suggestion
Chairs MUST be at least a "member" on our [contributor ladder](https://github.com/cilium/community/blob/main/CONTRIBUTOR-LADDER.md#organization-member) to be eligible to hold a leadership role within a SIG.
```

SIGs MAY prefer various levels of domain knowledge depending on the role. This should be documented.

> ### On-boarding and Off-boarding Leads

Reviewer 2: ```suggestion
### On-boarding and Off-boarding Chairs
```

Author: i thought we were sticking with the term "Leads"?

Reviewer 4: Yes, I thought we had decided on Lead. Either way we should get consistent in this doc

Reviewer 2: You're right, but everything below is about chairs so I got confused. Can you make it consistent?

Chairs MAY decide to step down at anytime and propose a replacement. Use lazy consensus amongst the other Chairs with fallback on majority vote to accept the proposal. The candidate SHOULD be supported by a majority of SIG contributors.
Chairs MAY select additional leads through a super-majority vote amongst Chairs. This SHOULD be supported by a majority of SIG contributors.

> 

Reviewer 2: ```suggestion
```

## Other Roles

> This governance document outlines the required roles for SIGs; however, SIGs are allowed to operate how they see fit outside of minimum governance requirements, including defining more roles to sustain the group. If a SIG needs to change the Chair position to include or remove duties, this needs to be approved by the committers. Newly created roles that don't assume any responsibility of Chairs should follow the governing processes in the SIGs charter.

Reviewer 2: ```suggestion
This governance document outlines the required roles for SIGs; however, SIGs are allowed to operate how they see fit outside of minimum governance requirements, including defining more roles to sustain the group. If a SIG needs to change the Chair position to include or remove duties, this needs to be approved by the committers. Newly created roles that don't assume any responsibility of Chairs should follow the governing processes in the SIG's charter.
```

# SIG Creation

## Goals

> The primary goal of the charters is to define the scope of the SIG within Kubernetes and how the SIG leaders exercise ownership of these areas by taking care of their responsibilities. A majority of the effort should be spent on these concerns.

Reviewer 2: ```suggestion
The primary goal of the charters is to define the scope of the SIG within Cilium and how the SIG leaders exercise ownership of these areas by taking care of their responsibilities. A majority of the effort should be spent on these concerns.
```

### SIG Charter Guide
All Cilium SIGs must define a charter defining the scope and governance of the SIG.
The scope must define what areas the SIG is responsible for directing and maintaining.
The governance must outline the responsibilities within the SIG as well as the roles owning those responsibilities.

> 

Reviewer 2: ```suggestion
```

### Steps to create a SIG charter

> 1. Copy the template into a new file under community/sig-YOURSIG/charter.md

Reviewer 2: there should be a link to the template

> 1. Read the Recommendations and requirements so you have context for the template

Reviewer 2: ```suggestion
1. Read the recommendations and requirements so you have context for the template
```

1. Fill out the template for your SIG
1. Update sigs.yaml with the individuals holding the roles as defined in the template.
1. Create a pull request with a draft of your charter.md and sigs.yaml changes. Communicate it within your SIG and get feedback as needed.
1. Send the SIG Charter out for review to the Cilium committers.
1. Once accepted, the committers will ratify the PR by merging it.

> 

Reviewer 2: ```suggestion
```

### Steps to update an existing SIG charter

> For significant changes, or any changes that could impact other SIGs, such as the scope, create a PR and send it to the committers for review with the subject: "SIG Charter Update: YOURSIG"

Reviewer 2: ```suggestion
For significant changes, or any changes that could impact other SIGs, such as the scope, create a PR and send it to the committers for review with the subject: "SIG Charter Update: YOURSIG".
```

For minor updates to that only impact issues or areas within the scope of the SIG the SIG Chairs should facilitate the change.

> SIG Charter approval process

Reviewer 2: ```suggestion

### SIG Charter approval process
```

> When introducing a SIG charter or modification of a charter the following process should be used. As part of this we will define roles for the OARP process (Owners, Approvers, Reviewers, Participants)

Reviewer 2: ```suggestion
When introducing a SIG charter or modification of a charter the following process should be used.
```

1. Identify a small set of people from the SIG to drive the changes. Most typically this will be the SIG chairs.
1. Work with the rest of the SIG in question to craft the changes. Make sure to keep the SIG in the loop as discussions progress with the committers.
1. Work with the committers to gain approval. This can simply be submitting a PR and sending a slack message. If more substantial changes are desired it is advisable to socialize those before drafting a PR.
1. The committers will be looking to ensure the scope of the SIG as represented in the charter is reasonable (and within the scope of Cilium) and that processes are fair.

> 

Reviewer 2: ```suggestion
```

# SIG Governance

To standardize Special Interest Group efforts, create maximum transparency, and route contributors to the appropriate SIG; SIGs should follow these guidelines:

- Have an approved Charter.
- Keep up-to-date meeting notes, linked from the SIG's page in the community repo.
- Ensure related work happens in a project-owned GitHub org and repository.
- Use the forums provided as the primary means of working, communicating, and collaborating, as opposed to private emails and meetings.
- Track and identify all CFPs and other project enhancements related to the SIG area.

## SIG Retirement
In the event that the SIG is unable to regularly establish consistent quorum or otherwise fulfill its Organizational Management responsibilities
- after 3 or more months it SHOULD be retired
- after 6 or more months it MUST be retired

> 

Reviewer 2: ```suggestion
```

## SIG Charter Template
### Scope
Include a 2-3 sentence summary of what work SIG TODO does. Imagine trying to explain your work to a colleague who is familiar with Cilium but not necessarily all of the internals.

#### In scope

> - Code, Binaries and Services

Reviewer 2: ```suggestion
**Code, Binaries and Services**
```

- list of what qualifies a piece of code, binary or service
- as falling into the scope of this SIG
- e.g. clis for working with Cilium APIs,
- CI for Cilium repos, etc

> - This is NOT a list of specific code locations,

Reviewer 2: ```suggestion
- This is NOT a list of specific code locations,

```

> Cross-cutting and Externally Facing Processes

Reviewer 2: ```suggestion
**Cross-cutting and Externally Facing Processes**
```

SIGs are designed to help contributors gain depth in a specific area of the Cilium project. Committers that are aspiring reviewers of the Cilium project can gain expertise by getting involved with a SIG and getting assigned PRs.

SIGs don't have any additional authority - they are expected to drive towards solutions that can be accepted by the project's committer community - ideally by consensus, and if necessary, a vote. 

A primary reason that SIGs exist is as forums for collaboration. Much work in a SIG should stay local within that SIG. However, SIGs must communicate in the open, ensure other SIGs and community members can find notes of meetings, discussions, designs, and decisions, and periodically communicate a high-level summary of the SIG's work to the community.

# SIG Roles

## SIG Lead

- Membership tracked in [sigs.yaml](./sigs.yaml)

> - MUST be a reviewer of one or more areas of the project, and should be a subject matter expert in their SIG area.

Reviewer: nit: Reordering so all "must" are together, "should" together and "may" together could help readers understand the requirements for operating a SIG.

- SHOULD drive charter changes (including creation) to get community buy-in but MAY delegate content creation to SIG contributors.
- MUST organize a method of keeping cadence (meetings, async channel in Cilium Slack, etc.), and make sure [sigs.yaml](./sigs.yaml) is up to date.
- SHOULD facilitate cadence but MAY delegate to other members or future Leads/Leads in training.

> - MUST ensure notes made available.

Reviewer 5: ```suggestion
- MUST ensure notes are made available.
```

> - MAY delegate the actual communication and creation of content to other contributors where appropriate.

Author: Good point! For the last line, I meant that when a Lead gets the channels available, they can give the more logistic tasks of "weekly updates" for example to someone else within the same SIG. For wording purposes, I'll combine 3 and 4 into 1 bullet point, something like:
> MAY delegate communication and note taking to other contributors in the same SIG where appropriate.

thoughts??

> - MAY delegate the actual communication and creation of content to other contributors where appropriate.

Reviewer: > MAY delegate communication and note taking to other contributors in the same SIG where appropriate.

Combining that way makes sense to me.

Thinking back over my confusion here and the wording, one thing I would like to encourage is that each SIG is driven by active participation from all the individuals in the SIG. I agree that it's completely fine and perhaps even recommended for SIG leaders to find ways to involve other members of the SIG in communications. Some people are more adept with communication so we should give them opportunities to contribute in that way. At the same time I think it's valuable to drive the sense of ownership within the SIG such that we don't have a "stadium" dynamic where the leader + delegates are "generating content" for a wider audience, but instead the group is driven collectively, more like a "club".

> - MAY delegate the actual communication and creation of content to other contributors where appropriate.

Reviewer: ```suggestion
- MAY delegate communication and note taking to other contributors in the same SIG where appropriate.
```

### Requirements
Leads MUST be at least a "reviewer" on our [contributor ladder](https://github.com/cilium/community/blob/main/CONTRIBUTOR-LADDER.md#reviewer) to be eligible to hold a leadership role within a SIG.

> SIGs MAY prefer various levels of domain knowledge depending on the role. This should be documented.

Reviewer 5: Please rephrase this part, as we discussed before :slightly_smiling_face:

> SIGs MAY prefer various levels of domain knowledge depending on the role. This should be documented.

Reviewer 5: Sorry, I don't understand this sentence.

Author: It's meant as leads can have different areas of expertise that's applicable to that sig, for example if sig-servicemesh had a lead in security, one in networking, etc

Reviewer 5: OK I get it, thank you. Can we rephrase it, please? Maybe something like:

```
Individual Leads MAY focus on specific subdomains for the SIG, such as security or networking in the context of a given SIG's scope. If some Leads chose to focus on particular domains of knowledge inside of the SIG, this SHOULD be documented.
```

Reviewer 5: This comment is marked as resolved, but has not been addressed

Author: oh my bad - sorry about that! Thanks Quentin 🥹

### On-boarding and Off-boarding Leads
Leads MAY decide to step down at anytime and propose a replacement. Use lazy consensus amongst the other Leads with fallback on majority vote to accept the proposal. The candidate SHOULD be supported by a majority of SIG contributors.

> Leads MAY select additional Leads through a super-majority vote amongst Leads. This SHOULD be supported by a majority of SIG contributors.

Reviewer: I guess "SIG contributors" isn't exactly defined here, but I'm assuming that we're thinking of active members of the organization who are engaging in creating and reviewing changes in the scope of the SIG?

Author: yes, ill add an explanation in that bullet point. Good catch 👍

> Leads MAY select additional Leads through a super-majority vote amongst Leads. This SHOULD be supported by a majority of SIG contributors (active members of the Cilium organization engaged with creating and reviweing changes within the SIG).

Reviewer 5: What's a “super-majority vote”?

Author: https://en.wikipedia.org/wiki/Supermajority#Two-thirds_vote 
Adding it as a link, good point to clarify 👍

Reviewer 5: Thanks! But don't you need to fix a threshold? “super-majority” itself just means that there must be a given percentage > 50% that votes in favour of the proposal, but it doesn't specify how much precisely? Or am I missing something?

Reviewer: I'd suggest just mentioning 2/3rds majority. That was my assumption when "super-majority" was mentioned.

Reviewer: Though actually let me maybe take a stab an at alternative proposal here and we can discuss from there (given my other feedback thread).

> Leads MAY select additional Leads through a super-majority vote amongst Leads. This SHOULD be supported by a majority of SIG contributors (active members of the Cilium organization engaged with creating and reviweing changes within the SIG).

Author: true. I do believe that SIGs would be a smaller group, and I like that it should be a discussion with the group. I also imagine that this consensus approach won't work - and we can change this to adopt to the way that works best/is observed 🤞

> Leads MAY select additional Leads through a super-majority vote amongst Leads. This SHOULD be supported by a majority of SIG contributors (active members of the Cilium organization engaged with creating and reviweing changes within the SIG).

Reviewer: Here's a alternate proposal for the text in this section:

```suggestion
Leads MAY nominate additional Leads for a SIG, or MAY choose to step down at any time. The change SHOULD be supported by a majority of SIG contributors (active members of the Cilium organization engaged with creating and reviweing changes within the SIG). All changes to SIG leadership MUST be conducted in consultation with the Committer group. 
```

In practice again I'd expect this is mainly conducted by proposing a change to the file in this repository. In the case that the last lead steps down, the committers may dissolve the SIG. (I wasn't sure whether to include dissolution in this section or not)

> Leads MAY select additional Leads through a super-majority vote amongst Leads. This SHOULD be supported by a majority of SIG contributors (active members of the Cilium organization engaged with creating and reviweing changes within the SIG).

Reviewer: ^ I don't feel super strongly that we need to state this is a Committer decision, but I think we should set the expectation that ultimately all of these changes are driven by the Committers executing their responsibility for [steering the project in a positive direction](https://github.com/cilium/community/blob/main/CONTRIBUTOR-LADDER.md#committer), and that SIGs do not preempt any of those rights or responsibilities. Though I don't anticipate us encountering any specific issues with this, given that SIG Leads are only required to be Reviewers and not Committers, Leads do not hold a stronger say over project affairs than the Committers.

## Other Roles
This governance document outlines the required roles for SIGs; however, SIGs are allowed to operate how they see fit outside of minimum governance requirements, including defining more roles to sustain the group. If a SIG needs to change the Lead position to include or remove duties, this needs to be approved by the committers. Newly created roles that don't assume any responsibility of Leads should follow the governing processes in the SIG's charter.

# SIG Creation

## Goals

> The primary goal of the charters is to define the scope of the SIG within Cilium and how the SIG leaders exercise ownership of these areas by taking care of their responsibilities. A majority of the effort should be spent on these concerns. SIG Leads are also reponsible for driving reviews for CFPs for their focus area, and have a responsibility to bring concerns to committers if they are unsure. 

Reviewer: What does this part mean?

>  A majority of the effort should be spent on these concerns.

I'm not sure I quite understand the expectations imposed on SIG leads specifically from this sentence. Maybe for a check: If we dropped this sentence, what meaning is lost?

Author: Meaning that what they outline in the charter will be their area of focus for that SIG, they can't just focus on another area if it doesnt relate back to that core mission

Reviewer 2: I think I agree with Joe. The word scope has a pretty clear meaning and the sentence doesn't add a lot.

Reviewer: > Meaning that what they outline in the charter will be their area of focus for that SIG, they can't just focus on another area if it doesnt relate back to that core mission

At the moment, I think the more likely scenario is that the same people are in multiple related SIGs driving any effort that intersects with both SIGs (example: policy and scale). Or the more common case in general is that something just falls under the project overall and isn't scoped within any particular SIG. I can see some potential in the idea you're looking to convey here, but my gut feeling is that it's not something we need to worry about with the current size of the project. I'd suggest dropping this specific sentence for now, then we can always keep an eye out on how this is working and amend the wording in future if we feel this needs stronger emphasis.

> The primary goal of the charters is to define the scope of the SIG within Cilium and how the SIG leaders exercise ownership of these areas by taking care of their responsibilities. A majority of the effort should be spent on these concerns. SIG Leads are also reponsible for driving reviews for CFPs for their focus area, and have a responsibility to bring concerns to committers if they are unsure. 

Reviewer: What does "driving reviews" mean?

Currently we primarily rely upon GitHub automated review assignment for PR review (at least for the most popular repositories). While this isn't a perfect system, I think it can help with distributing the workload.

Probably as a fallback it makes sense for the Lead to keep an eye on the reviews, but in a well-functioning SIG I would hope that we are able to distribute the load between the folks in the team to ensure we don't overload the Leads.

(This may be something we end up just settling on some wording for, then work with the initial SIGs to clarify / document expectations).

Author: fair, i think driving review overall means that a CFP will be responded to in a timely manner, something hands off for the leads ultimately (gentle pokes to whoever is reviewing within their SIG), making sure conversation doesn't go stale? I do realize that adds a lot of glue work for the leads though, and i dont want the work to grow unmeasured.

Reviewer 2: I think I agree with Joe. The word scope has a pretty clear meaning and the sentence doesn't add a lot.

Reviewer 2: Right now we don't have automated assignment for the CFP repo. I think this is mean to say that the SIG Leaders should try to make sure that the CFPs in their area get reviewed. I've suggested some updated wording below.

> The primary goal of the charters is to define the scope of the SIG within Cilium and how the SIG leaders exercise ownership of these areas by taking care of their responsibilities. SIG Leads and SIGs are responsible for reviewing CFPs in their focus area, and have a responsibility to bring concerns to committers if they are unsure. 

Reviewer 5: ```suggestion
The primary goal of the charters is to define the scope of the SIG within Cilium and how the SIG leaders exercise ownership of these areas by taking care of their responsibilities. SIG Leads and SIGs are responsible for reviewing CFPs in their focus area, and have a responsibility to bring concerns to Committers if they are unsure. 
```

Here and elsewhere - “Committers” is a specific title in this context

> The primary goal of the charters is to define the scope of the SIG within Cilium and how the SIG leaders exercise ownership of these areas by taking care of their responsibilities. A majority of the effort should be spent on these concerns. SIG Leads are also reponsible for driving reviews for CFPs for their focus area, and have a responsibility to bring concerns to committers if they are unsure. 

Reviewer 2: ```suggestion
The primary goal of the charters is to define the scope of the SIG within Cilium and how the SIG leaders exercise ownership of these areas by taking care of their responsibilities. SIG Leads are also responsible for driving reviews for CFPs for their focus area, and have a responsibility to bring concerns to committers if they are unsure. 
```

### SIG Charter Guide
All Cilium SIGs must define a charter defining the scope and governance of the SIG.
The scope must define what areas the SIG is responsible for directing and maintaining.
The governance must outline the responsibilities within the SIG as well as the roles owning those responsibilities.

> ### Steps to create a SIG charter

Reviewer 5: Should there be a subsection mentioning how a SIG might be dissolved (other than by retirement)? Pessimistic example:

- Aspirant Lead open PR to propose new SIG
- Some Committers merge quickly
- Other Committers raise concerns
- Committers vote against the new SIG

Technically the new SIG has been created already. Similarly, if for some unlikely reason a SIG becomes counter-productive or in contradiction with the CoC, it might be necessary to dissolve it. Should there be a line to mention that the Committers can decide to dissolve a SIG under extraordinary circumstances?

Author: hmmm, I personally don't see the need to add a new section, but adding a sentence to the SIG retirement section to  outline a SIG may be dissolved if it doesn't abide by Cilium's CoC?

cc @xmulligan thoughts??

Reviewer 5: Yeah doesn't have to be a full subsection, just a sentence is enough as long as it's mentioned somewhere, I'd say

Reviewer: Yeah I think it should be clear that SIGs only exist at the direction of the Committers who ultimately decide everything in the project. Just as we are establishing SIGs now, we could dissolve them in future for a range of reasons.

Reviewer 5: You phrased it much better than I did :)

1. Fill out the template for your SIG
1. Update sigs.yaml with the individuals holding the roles as defined in the template.
1. Create a pull request with a draft of your charter.md and sigs.yaml changes. Communicate it within your SIG and get feedback as needed.
1. Send the SIG Charter out for review to the Cilium committers.
1. Once accepted, the committers will ratify the PR by merging it.

### Steps to update an existing SIG charter
For significant changes, or any changes that could impact other SIGs, such as the scope, create a PR and send it to the committers for review with the subject: "SIG Charter Update: YOURSIG".
For minor updates to that only impact issues or areas within the scope of the SIG the SIG Leads should facilitate the change.

### SIG Charter approval process
When introducing a SIG charter or modification of a charter the following process should be used.

1. Identify a small set of people from the SIG to drive the changes. Most typically this will be the SIG Leads.
1. Work with the rest of the SIG in question to craft the changes. Make sure to keep the SIG in the loop as discussions progress with the committers.

> 1. Work with the committers to gain approval. This can simply be submitting a PR and sending a slack message. If more substantial changes are desired it is advisable to socialize those before drafting a PR.

Reviewer: I think this is probably implied, but "socialize" is a bit vague. An example might help.

```suggestion
1. Work with the committers to gain approval. This can simply be submitting a PR and sending a slack message. If more substantial changes are desired it is advisable to socialize those before drafting a PR, for instance by connecting with a committer to act as a mentor and sponsor for this process.
```

1. The committers will be looking to ensure the scope of the SIG as represented in the charter is reasonable (and within the scope of Cilium) and that processes are fair.

# SIG Governance

> To standardize Special Interest Group efforts, create maximum transparency, and route contributors to the appropriate SIG; SIGs should follow these guidelines:

Reviewer 5: ```suggestion
To standardize SIGs efforts, create maximum transparency, and route contributors to the appropriate SIG, SIGs should follow these guidelines:
```

- Have an approved Charter.
- Keep up-to-date meeting notes, linked from the SIG's page in the community repo.
- Ensure related work happens in a project-owned GitHub org and repository.

> - Use the forums provided as the primary means of working, communicating, and collaborating, as opposed to private emails and meetings.

Reviewer: "forums provided" implies the reader knows already what those forums are. Might be worth iterating that these are the public open spaces with first preference of GitHub, secondary preference of Slack and third preference of recorded group meetings. 

(We could debate 1st/2nd/3rd but I ordered those by accessibility and permanence of communication)

# Cilium SIGS

The Cilium project has Special Interest Groups, or SIGs, with a common purpose of advancing the project with respect to a specific topic, such as network policy or documentation. Our goal is to enable a distributed decision structure and code ownership, as well as providing focused forums for getting work done, making decisions, and onboarding new contributors. 

Areas covered by SIGs may be vertically focused on particular components or functions, cross-cutting/horizontal, spanning many/all functional areas of the project, or in support of the project itself. Examples:

- Policy
- Scalability
- Docs

> Each SIG must have a charter that specifies its scope (topics, subsystems, code repos, and/or directories), responsibilities, and areas of authority. See the SIG charter process for details on how charters are managed. SIGs are relatively free to customize or change how they operate, within some broad guidelines and constraints imposed by cross-project processes (e.g., the release process) and assets (e.g., the Cilium repo).

Reviewer: This fragment conflicts with a couple of paragraphs down.

```suggestion
Each SIG must have a charter that specifies its scope (topics, subsystems, code repos, and/or directories), and responsibilities. See the SIG charter process for details on how charters are managed. SIGs are relatively free to customize or change how they operate, within some broad guidelines and constraints imposed by cross-project processes (e.g., the release process) and assets (e.g., the Cilium repo).
```

I'd be open to exploring changes in authority in future to help scale the project, but for now I think it makes sense to keep the authority with the committers, partially delegated to reviewers. Then SIGs may facilitate communication amongst members that already have authority through the regular process, but community members do not gain or exercise any specific authority by virtue of SIG participation.

SIGs are designed to help contributors gain depth in a specific area of the Cilium project. Committers that are aspiring reviewers of the Cilium project can gain expertise by getting involved with a SIG and getting assigned PRs.

SIGs don't have any additional authority - they are expected to drive towards solutions that can be accepted by the project's committer community - ideally by consensus, and if necessary, a vote. 

> A primary reason that SIGs exist is as forums for collaboration. Much work in a SIG should stay local within that SIG. However, SIGs must communicate in the open, ensure other SIGs and community members can find notes of meetings, discussions, designs, and decisions, and periodically communicate a high-level summary of the SIG's work to the community.

Reviewer: nit: wording
```suggestion
A primary reason that SIGs exist is as forums for collaboration. Much of the work in a SIG should stay local within that SIG. However, SIGs must communicate in the open, ensure other SIGs and community members can find notes of meetings, discussions, designs, and decisions, and periodically communicate a high-level summary of the SIG's work to the community.
```

Author: haha good catch, sounds like meme text :doge:

Reviewer: Much work, very collaboration. Wow.

# SIG Roles

## SIG Lead

> - Membership tracked in [sigs.yaml](./sigs.yaml)

Reviewer: nit: The other statements in this section are all prefaced with MUST/SHOULD/MAY. Is this specific line relating to leads or is it a more general statement that could live elsewhere in the doc?

Author: fair point, ill add in MUST

> - MUST be an odd number of SIG Leads in a group.

Reviewer 6: This is an odd requirement :wink: 
I would say that for example 2+ or 1+ is probably better. 
Odd number would mean that, for example, sig starts with 1 lead, and can have 3 only if 2 new leads are added at the same time.

Reviewer 2: I agree, I would remove this for now. I think we can revisit again if/when it becomes a problem. Also both of our current sigs have two leaders

Author: noted! will remove - this was following a recommendation to avoid stalemates 😅

Reviewer: Generally speaking we aim to work on consensus and not tie-breaking votes at the SIG level. If necessary we can escalate these to the wider committer group or ultimately maintainers.

- MUST ensure notes are made available.
- MUST organize a method of keeping cadence (meetings, async channel in Cilium Slack, etc.), and make sure [sigs.yaml](./sigs.yaml) is up to date.
- SHOULD facilitate cadence but MAY delegate to other members or future Leads/Leads in training.
- SHOULD drive charter changes (including creation) to get community buy-in but MAY delegate content creation to SIG contributors.
- MAY delegate communication and note taking to other contributors in the same SIG where appropriate.

### Requirements
Leads MUST be at least a "reviewer" on our [contributor ladder](https://github.com/cilium/community/blob/main/CONTRIBUTOR-LADDER.md#reviewer) to be eligible to hold a leadership role within a SIG.
SIGs MAY prefer various levels of domain knowledge depending on the role. This should be documented.

### On-boarding and Off-boarding Leads

> Leads MAY decide to step down at anytime and propose a replacement. Use lazy consensus amongst the other Leads with fallback on majority vote to accept the proposal. The candidate SHOULD be supported by a majority of SIG contributors (active members of the Cilium organization engaged with creation and reviews in the scope of the SIG).

Reviewer: I think we can drop this line? The following line covers the same information but with a conflicting approach.
```suggestion

```

# Cilium SIGS

The Cilium project has Special Interest Groups, or SIGs, with a common purpose of advancing the project with respect to a specific topic, such as network policy or documentation. Our goal is to enable a distributed decision structure and code ownership, as well as providing focused forums for getting work done, making decisions, and onboarding new contributors. 

Areas covered by SIGs may be vertically focused on particular components or functions, cross-cutting/horizontal, spanning many/all functional areas of the project, or in support of the project itself. Examples:

- Policy
- Scalability
- Docs

> Each SIG must have a charter that specifies its scope (topics, subsystems, code repos, and/or directories), responsibilities and areas of authority, for example, a SIG Lead. See the SIG charter process for details on how charters are managed. SIGs are relatively free to customize or change how they operate, within some broad guidelines and constraints imposed by cross-project processes (e.g., the release process) and assets (e.g., the Cilium repo).

Reviewer 2: ```suggestion
Each SIG must have a charter that specifies its scope (topics, subsystems, code repos, and/or directories), responsibilities, and areas of authority. See the SIG charter process for details on how charters are managed. SIGs are relatively free to customize or change how they operate, within some broad guidelines and constraints imposed by cross-project processes (e.g., the release process) and assets (e.g., the Cilium repo).
```

SIGs are designed to help contributors gain depth in a specific area of the Cilium project. Committers that are aspiring reviewers of the Cilium project can gain expertise by getting involved with a SIG and getting assigned PRs.

SIGs don't have any additional authority - they are expected to drive towards solutions that can be accepted by the project's committer community - ideally by consensus, and if necessary, a vote. 

A primary reason that SIGs exist is as forums for collaboration. Much work in a SIG should stay local within that SIG. However, SIGs must communicate in the open, ensure other SIGs and community members can find notes of meetings, discussions, designs, and decisions, and periodically communicate a high-level summary of the SIG's work to the community.

> # SIG Roles

Reviewer 2: ```suggestion
# SIG Roles

```

## SIG Lead

- Membership tracked in [sigs.yaml](./sigs.yaml)
- MUST be an odd number of SIG Leads in a group.
- MUST be a reviewer of one or more areas of the project, and should be a subject matter expert in their SIG area.
- SHOULD drive charter changes (including creation) to get community buy-in but MAY delegate content creation to SIG contributors.
- MUST organize a method of keeping cadence (meetings, async channel in Cilium Slack, etc.), and make sure [sigs.yaml](./sigs.yaml) is up to date.
- SHOULD facilitate cadence but MAY delegate to other members or future Leads/Leads in training.
- MUST ensure notes made available.
- MAY delegate the actual communication and creation of content to other contributors where appropriate.

### Requirements

> Leads MUST be at least a "member" on our [contributor ladder](https://github.com/cilium/community/blob/main/CONTRIBUTOR-LADDER.md#organization-member) to be eligible to hold a leadership role within a SIG.

Reviewer 2: ```suggestion
Leads MUST be at least a "reviewer" on our [contributor ladder](https://github.com/cilium/community/blob/main/CONTRIBUTOR-LADDER.md#reviewer) to be eligible to hold a leadership role within a SIG.
```

For minor updates to that only impact issues or areas within the scope of the SIG the SIG Leads should facilitate the change.

### SIG Charter approval process
When introducing a SIG charter or modification of a charter the following process should be used.

1. Identify a small set of people from the SIG to drive the changes. Most typically this will be the SIG Leads.
1. Work with the rest of the SIG in question to craft the changes. Make sure to keep the SIG in the loop as discussions progress with the committers.
1. Work with the committers to gain approval. This can simply be submitting a PR and sending a slack message. If more substantial changes are desired it is advisable to socialize those before drafting a PR.
1. The committers will be looking to ensure the scope of the SIG as represented in the charter is reasonable (and within the scope of Cilium) and that processes are fair.

# SIG Governance

To standardize Special Interest Group efforts, create maximum transparency, and route contributors to the appropriate SIG; SIGs should follow these guidelines:

- Have an approved Charter.

> - Keep up-to-date meeting notes, linked from the SIG's page in the community repo.

Reviewer 2: Do you think this should be in the README or the sigs.yaml?

k8s does sigs.yaml for contact info, meeting links, time, ect. https://github.com/kubernetes/community/blob/master/sigs.yaml#L192

Might be easier to consume in a README though?

> - Keep up-to-date meeting notes, linked from the SIG's page in the community repo.

Author: fair- i still think they should be in both. Having the sigs.yaml at the root will be good for a quick search in one file
